tests/closure/tdonotation.nim

```diff
@@ -0,0 +1,47 @@
+discard """
+output: '''
+click at 10,20
+lost focus 2
+registered handler for UserEvent 1
+registered handler for UserEvent 3'''
+"""
+
+import future
+
+type
+  Button = object
+  Event = object
+    x, y: int
+
+proc onClick(x: Button, handler: proc(x: Event)) =
+  handler(Event(x: 10, y: 20))
+
+proc onFocusLost(x: Button, handler: proc()) =
+  handler()
+
+proc onUserEvent(x: Button, eventName: string, handler: proc) =
+  echo "registered handler for ", eventName
+
+var b = Button()
+
+b.onClick do (e: Event):
+  echo "click at ", e.x, ",", e.y
+
+when false:
```

Comment:
I'm not sure this syntax is a good idea. We shouldn't conflate template invocation syntax with procedure invocation syntax too much, especially since this syntax will only be useful for procedures that take no parameters.